### PR TITLE
[coq] Default for mode vo/native according to coqc configuration

### PR DESCRIPTION
**Pull request by proux01**
This is probably not perfect but is already making dune work with the coq-native OPAM package.

### CODE REVIEW

**Comment by ejgallego** [comment]
I'm afraid this is unsound at several levels; think that coqc may be produced by a rule itself and not present in the path.

Actually in Dune 3.0 you may be able to write this correctly as the rules may depend on the build action of `coqc` itself, but on the other hand it is IMHO not worth it as starting with 8.14 we have a much simpler setup:

- setup the native rules unconditionally (relying on coqnative) [which will actually run in parallel, thus a large gain]
- have the native package install target pull the native files into scope

Thus, no conditionals anymore, other than indeed the corresponding top level alias or install file which can be triggered by the user at will.

**Comment by ejgallego** [comment]
Would we like to solve this directly, then we would have to compute the mode inside the build monad, so we should obtain native mode as a build rule [which is then correctly tracked by the dependencies].

Note also that `System` is frowned upon as we need dune to be portable among systems, but in this case you can just cache the output in a file and parse it [tho one has to be careful as not to overaproximate the flag with the full file, as that would trigger spurious rebuilds]

As I mentioned, for native this is IMHO overkill, but could be cool as to add support for other stuff such as a `%{coq:version}` variable.

**Comment by proux01** [comment]
Thanks for your comments and ideas. I agree this PR is way too broken. I also agree coqnative and the split package solution is the long term way to go. Unfortunately, this still requires some work on the opam repo side that noone is doing now (I may take a look but probably not soon). Meanwhile, any package switching to dune breaks coq-native (we already have the issue with multinomials).

Here is a simpler idea : make the default value configurable at dune build time, just like for coqc. Things would then remain exactly as now, except when the coq-native package is installed in which case (mode native) would become the default. Since coqc is built the exact same way, dune and coqc should remain in sync. That could constitue a reasonnable short term solution until we actually switch to coqnative.

**Comment by rgrinberg** [comment]
Closing as this PR was mostly useful for discussion. It's not permitted to run external commands in dune this way.

**Comment by proux01** [comment]
@rgrinberg sure but the problem remains: dune is stil incompatible with the coq-native OPAM package. This is IMHO the main obstacle to the use of dune to build any Coq library that could potentially be used with native (which, in practice, means virtually any Coq library). Or at least not without painful hacks and adding a confiure phase to tweak the dune files (see https://github.com/math-comp/multinomials ). I was told by @ejgallego this could be fixed with Dune 3.0 but since I don't know how, I'm unable to do it myself. Unfortunately today, due to this issue, we have no other solution than to recommend *not* to use dune to build Coq libraries.

So to sum up, the PR is indeed wrong but the problem it adressed remains.

**Comment by Alizter** [comment]
@proux01 I added the infrastructure for Dune to see the Coq configuration in 3.4. Notably we can see if native is configured. It should be straightforward to use this.

**Comment by proux01** [comment]
@Alizter great! I should have a look. Would you have a more precise pointer to the feature?

**Comment by Alizter** [comment]
@proux01  https://github.com/ocaml/dune/pull/6049

**Comment by Alizter** [comment]
@proux01 If you open an issue detailing how Dune should act depending on the output of the native config var from Coq I can implement it (maybe even this week). We should be releasing 3.6 soon too FYI.
